hack/make-rules/test-integration.sh

```diff
@@ -94,9 +87,11 @@ runTests() {
 checkEtcdOnPath() {
   kube::log::status "Checking etcd is on PATH"
   which etcd && return
+  . hack/install-etcd.sh
```

Comment:
I couldn't get integration tests to run.  Without this it says something like "you can run hack/install-etcd.sh" but that doesn't actually work.  I need to fix that up

Reply:
`third_party/etcd` must be in the PATH. I suppose the help message could be improved to state that explicitly.

I have that and then it works without this change:
```console
$ make test-integration
+++ [0913 14:56:17] Checking etcd is on PATH
/nvme/gopath/src/k8s.io/kubernetes/third_party/etcd/etcd
+++ [0913 14:56:17] Starting etcd instance
etcd --advertise-client-urls http://127.0.0.1:2379 --data-dir /tmp/tmp.PqLAFW7STQ --listen-client-urls http://127.0.0.1:2379 --log-level=warn 2> "/dev/null" >/dev/null
Waiting for etcd to come up.
+++ [0913 14:56:18] On try 1, etcd: : {"health":"true","reason":""}
{"header":{"cluster_id":"14841639068965178418","member_id":"10276657743932975437","revision":"2","raft_term":"2"}}Periodically scraping etcd to /tmp/test.5exoVn/etcd-scrapes .
+++ [0913 14:56:18] Running integration test cases
make[1]: Entering directory '/nvme/gopath/src/k8s.io/kubernetes'
+++ [0913 14:56:18] Set GOMAXPROCS automatically to 36
+++ [0913 14:56:18] Normalizing Go targets
+++ [0913 14:56:19] Running tests without code coverage 
?   	k8s.io/apiextensions-apiserver/test/integration/fixtures	[no test files]
?   	k8s.io/apiextensions-apiserver/test/integration/storage	[no test files]
?   	k8s.io/kubernetes/test/integration	[no test files]
I0913 14:56:53.465693  146868 etcd.go:73] etcd already running at http://127.0.0.1:2379
?   	k8s.io/kubernetes/test/integration/apiserver/cel/authorizerselector	[no test files]
?   	k8s.io/kubernetes/test/integration/authutil	[no test files]
?   	k8s.io/kubernetes/test/integration/benchmark/extractlog	[no test files]
?   	k8s.io/kubernetes/test/integration/benchmark/jsonify	[no test files]
I0913 14:56:56.725931  147094 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.728872  147098 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.729463  147113 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.729534  147090 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.730951  147089 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.731325  147095 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.732114  147091 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.734625  147109 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.734872  147112 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.736037  147105 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.737225  147103 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.736960  147086 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.740642  147104 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.741558  147107 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.742152  147087 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.742499  147100 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.744239  147110 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.751267  147088 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.749411  147097 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.748159  147101 i18n.go:157] Setting language to en_US
I0913 14:56:56.760796  147099 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.760947  147085 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.762197  147092 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.762621  147093 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.774763  147101 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.774781  147111 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:56:56.799436  147102 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:57:01.768693  148277 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:57:02.572990  148348 etcd.go:73] etcd already running at http://127.0.0.1:2379
I0913 14:57:02.576836  148347 etcd.go:73] etcd already running at http://127.0.0.1:2379
ok  	k8s.io/kubernetes/test/integration/apiserver/cel/authorizerselector/selectorenabled	5.938s
...
```

I find etcd.go too noisy here, but that's a different story.

A bit more on topic is that Go now warns about `no test files` in packages like `k8s.io/kubernetes/test/integration`. It didn't do that before.

Reply:
I really think the integration tests should handle setting up the etcd in path (maybe with some override option) without having to call out to bash etc. (e.g. https://github.com/kubernetes/registry.k8s.io/blob/cdd7484e9c1edd2acb1130cc654d5ac232808fb2/cmd/archeio/internal/e2e/e2e_containerd_linux_test.go)

Reply:
This would promote `hack/install-etcd.sh` from "here's a suggestion how you might want to install etcd" to "this is how etcd gets installed unless a user did something else". I don't have any objection against that, I'm just double-checking whether we are okay with that.

Technically, the Go binary would have to check for etcd and then invoke the script. Seems doable. We might have to be careful about races when different binaries do that concurrently.